File: cmd/models/notification.go
package models

import "time"

type Notification struct {
	ID          string    `json:"id"`
	UserID      string    `json:"userId"`
	Type        string    `json:"type"`
	Title       string    `json:"title"`
	Message     string    `json:"message"`
	ReferenceID *string   `json:"referenceId"`
	IsRead      bool      `json:"isRead"`
	CreatedAt   time.Time `json:"createdAt"`
}

File: app/src/types/notification.ts
export type NotificationType =
  | 'assigned'
  | 'delegated'
  | 'deadline'
  | 'comment'
  | 'report_request'
  | 'completed'
  | 'overdue'

export interface Notification {
  id: string
  userId: string
  type: NotificationType
  title: string
  message: string
  referenceId: string | null
  isRead: boolean
  createdAt: string
}
